.github/copilot-instructions.md

```diff
@@ -0,0 +1,145 @@
+- Code examples should be included for key functionality
+- Follow the specific format for injecting code snippets in README files:
+
+<pre>
+```java readme-sample-sampleName
+  Code snippet
+```
+</pre>
+
+## Azure SDK Guidelines
+
+Azure client libraries for Java should adhere strictly to these guidelines.
+
+Core Principles:
+
+- Be idiomatic, consistent, approachable, diagnosable, and dependable.
+- Use natural Java patterns and follow modern Java practices (try-with-resources, CompletableFuture, Streams, Optional).
+
+API Design:
+
+- Create service client classes (with "Client" suffix) with fluent builder patterns.
+- Use options bags (e.g., `<MethodName>Options`) for additional parameters.
+- Follow standard verbs (create, upsert, get, delete, etc.).
+- Provide both synchronous and asynchronous clients with the async version suffixed with "AsyncClient".
+
+Implementation:
+
+- Follow semver guidelines. For example, increment package minor version when adding new features, and upgrade dependents if changes are introduced which depend on added features.
+- Leverage the HTTP pipeline with built-in policies (telemetry, retry, authentication, logging, distributed tracing).
+- Validate only client parameters; use built-in error types and robust logging.
+- Use core packages like `azure-core` and follow consistent patterns for authentication, logging, and tracing.
+
+Prioritize Java-specific practices over general rules when conflicts occur.
+
+When possible, refer to the Azure SDK for Java Design Guidelines for specific examples and best practices. Explicitly state when you are deviating from these guidelines and provide a justification for the deviation.
+
+## Pull Request Guidelines
+
+- Ensure all tests pass
+- Follow the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md)
+- Include appropriate documentation
+- Include tests that cover your changes
+- Update CHANGELOG.md with your changes
+- Provide a proper description of the pull request to document the changes in the PR. The description should include:
+  - A summary of the changes made.
+  - The reason for the changes.
+  - Any relevant issue numbers.
+  - Instructions on how to verify the changes.
+  - Any additional context or information that reviewers should be aware of.
+
+## Release Process
+
+- Version numbers follow [Semantic Versioning](https://semver.org/)
+- Libraries are released to Maven Central
+- Beta releases are denoted with `-beta.N` suffix
+
+## Troubleshooting
+
+- Enable client logging for debugging
+- Use HTTP pipeline policies to customize behavior
+- Refer to service-specific troubleshooting guides
+
+## Third-party Dependencies
+
+- External dependencies should be referenced from `external_dependencies.txt`
+- Third-party libraries should only be included when necessary
+- Prefer OSI-approved licensed dependencies
+
+## Support Channels
+
+When facing issues, direct users to:
```

Comment:
add https://github.com/Azure/azure-sdk-for-java/blob/main/SUPPORT.md ?

Reply:
good idea, updated!